yaml_edit/lib/src/editor.dart

```diff
@@ -572,10 +551,7 @@ class YamlEditor {
   /// tree is equal to our expectations by deep equality of values. Throws an
   /// [AssertionError] if the two trees do not match.
   void _performEdit(
-      SourceEdit edit, Iterable<Object> path, YamlNode expectedNode) {
-    ArgumentError.checkNotNull(edit, 'edit');
-    ArgumentError.checkNotNull(path, 'path');
-
+      SourceEdit edit, Iterable<Object?> path, YamlNode? expectedNode) {
```

Comment:
Can `expectedNode` be `null`? Shouldn't callers be giving a `YamlNode` where `.value == null` ?
(I'm just asking, but it looks like that it's only `remove(` that might have to do this, right?

Reply:
Yes and it's for that reason that I changed the signature of `_performEdit()` and `_deepModify()` to accept nullable `expectedNode`. This is similar to the issue I had with `edit` value where I wasn't sure what the default value should be.

When I changed it to:

```dart
  YamlNode remove(Iterable<Object?> path) {
    SourceEdit? edit;
    var expectedNode = wrapAsYamlNode(null);
    final nodeToRemove = _traverse(path, checkAlias: true);

    if (path.isEmpty) {
...
```
...and changed the the rest of the methods to expect a non-nullable `expectedNode`, there were no errors when running tests. However, I'm still unsure what `expectedNode` does in this context so I don't know if this is the correct approach.